custom_components/foxess_modbus/entities/modbus_sensor.py

```diff
@@ -81,3 +92,7 @@ def state_class(self) -> SensorStateClass:
     @property
     def should_poll(self) -> bool:
         return False
+
+    def _validate(self, rules, value) -> bool:
+        """Validate against a set of rules"""
+        return all(rule.validate(value) for rule in rules)
```

Comment:
`rules` can be `None`, according to its type annotation?

Reply:
Thought i'd set it to `field(default_factory=list)

Reply:
Right, but someone can assign `validate = None`, which is valid according to the type hint, but that will make this function crash. mypi would be moaning at you here :)

Reply:
(I'd just remove `| None` from the type hint if you don't intend to allow `None`)